scripts/check_code_style.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2019 Advanced Micro Devices, Inc. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+GFX reconstruct check code style script
+'''
+
+import subprocess
+import sys
+import argparse
+
+def check_code_style(file):
+    """
+    Run clang format diff on the changed source file(s)
+    against current branch. If branch retrieved failed, 
+    fall back to diff against master branch.
+    :raises Exception: when formatting errors found
+    """
+    git_branch = subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"]).strip()
+    if git_branch == b'':
+        git_branch = "master"
+    cmd = ["git", "diff", "-U0", str(git_branch, 'utf-8'),
+           "--"] + file + ["|", "python", "clang-format-diff.py","-p1","-style=file"]
+    try:
+        cmd_output = subprocess.check_output(cmd)
```

Comment:
If you're using a shell pipe, you should use a string command rather than an array:
```
    cmd = 'git diff -U0 {} -- {} | python clang-format-diff.py -p1 -style=file'.format(str(git_branch, 'utf-8'), file)
    cmd_output = subprocess.check_output(cmd, shell=True)
```
I don't recommend using `file` as a variable (it's also a keyword).  I really don't recommend using `shell=True` either, except as a last resort or in a throwaway script.  There are other ways to make this happen.

Reply:
This is resolved in 15c63bf with the command split into two separate subprocess calls.